contracts/utility/CarbonBatcher.sol

```diff
@@ -0,0 +1,224 @@
+        ICarbonController _carbonController,
+        IVoucher _voucher
+    ) validAddress(address(_carbonController)) validAddress(address(_voucher)) {
+        carbonController = _carbonController;
+        voucher = _voucher;
+        _disableInitializers();
+    }
+
+    /**
+     * @dev fully initializes the contract and its parents
+     */
+    function initialize() public initializer {
+        __CarbonBatcher_init();
+    }
+
+    // solhint-disable func-name-mixedcase
+
+    /**
+     * @dev initializes the contract and its parents
+     */
+    function __CarbonBatcher_init() internal onlyInitializing {
+        __Upgradeable_init();
+    }
+
+    /**
+     * @inheritdoc Upgradeable
+     */
+    function version() public pure virtual override(Upgradeable) returns (uint16) {
+        return 1;
+    }
+
+    /**
+     * @notice creates several new strategies, returns the strategies id's
+     *
+     * requirements:
+     *
+     * - the caller must have approved the tokens with assigned liquidity in the orders
+     */
+    function batchCreate(
+        StrategyData[] calldata strategies
+    ) external payable greaterThanZero(strategies.length) nonReentrant returns (uint256[] memory) {
+        uint256[] memory strategyIds = new uint256[](strategies.length);
+        uint256 txValueLeft = msg.value;
+
+        // extract unique tokens and amounts
+        (Token[] memory uniqueTokens, uint256[] memory amounts) = _extractUniqueTokensAndAmounts(strategies);
+        // transfer funds from user for strategies
+        for (uint256 i = 0; i < uniqueTokens.length; i = uncheckedInc(i)) {
+            Token token = uniqueTokens[i];
+            uint256 amount = amounts[i];
+            if (token.isNative()) {
+                if (txValueLeft < amount) {
+                    revert InsufficientNativeTokenSent();
+                }
+                txValueLeft -= amount;
+                continue;
+            }
+            token.safeTransferFrom(msg.sender, address(this), amount);
+            _setCarbonAllowance(token, amount);
+        }
+
+        // create strategies and transfer nfts to user
+        for (uint256 i = 0; i < strategies.length; i = uncheckedInc(i)) {
+            // get tokens for this strategy
+            Token[2] memory tokens = strategies[i].tokens;
+            Order[2] memory orders = strategies[i].orders;
+            // if any of the tokens is native, send this value with the create strategy tx
+            uint256 valueToSend = 0;
+            if (tokens[0].isNative()) {
+                valueToSend = orders[0].y;
+            } else if (tokens[1].isNative()) {
+                valueToSend = orders[1].y;
+            }
+
+            // create strategy on carbon
+            strategyIds[i] = carbonController.createStrategy{ value: valueToSend }(tokens[0], tokens[1], orders);
+            // transfer nft to user
+            voucher.safeTransferFrom(address(this), msg.sender, strategyIds[i], "");
```

Comment:
Hmmm.. maybe `createStrategy` can receive an `receipient` or `owner` argument and mint the voucher to it directly? This will prevent the additional transfer.